carddav/elements.go

```diff
@@ -211,3 +211,33 @@ func (r *reportReq) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
 
 	return d.DecodeElement(v, &start)
 }
+
+type rtdef struct {
+	IsCollection  bool
+	IsAddressBook bool
+}
+
+type resourcetype struct {
```

Comment:
We already have `internal.ResourceType`, maybe it's good enough for our purposes?

Reply:
Thanks for reminding me of that! I updated the second commit, it looks much better now